examples/pipelines/commoncrawl/components/filter_webpage_urls/src/main.py

```diff
@@ -0,0 +1,54 @@
+import os
+import logging
+from typing import List
+
+import dask.dataframe as dd
+from urllib.parse import urlparse
+
+from fondant.component import DaskTransformComponent
+from fondant.executor import DaskTransformExecutor
+
+from utils.file_utils import download_tar_file, get_urls_from_file
+
+logger = logging.getLogger(__name__)
+
+BLACKLIST_DIR = "/tmp/blacklists"
+BLACKLIST_URL = "http://dsi.ut-capitole.fr/blacklists/download/blacklists.tar.gz"
+
+
+def get_blacklisted_domains(blacklist_dir: str, categories: List[str]) -> List[str]:
+    result = []
```

Comment:
Could you use a `set` instead of `list`? Should be faster to check if an element is in the list. Especially if the collection contains millions of entries.